files/initramfs-tools/union-mount.j2

```diff
@@ -218,8 +218,10 @@ mkdir -p ${rootmnt}/boot
 mkdir -p ${rootmnt}/host/$image_dir/boot
 mount --bind ${rootmnt}/host/$image_dir/boot ${rootmnt}/boot
 
+{% if ARCH | default("x86_64") != "armhf" %}
 ## Mount the /tmp directory as tmpfs
 mount -t tmpfs -o rw,nosuid,nodev,size=25% tmpfs ${rootmnt}/tmp
+{% endif %}
```

Comment:
Instead of all these changes, `DPKG_ARCH` can be used (please test with this, I'm not 100% certain `armhf` is the correct string.

```suggestion
if [ $DPKG_ARCH != "armhf" ]; then
    ## Mount the /tmp directory as tmpfs
     mount -t tmpfs -o rw,nosuid,nodev,size=25% tmpfs ${rootmnt}/tmp
 fi
```

Better than this, though, is if we can do this based on available memory, that would be the best, because that is the underlying reason that this change is being made.

Reply:
Hi @saiarcot895, the j2 cli take `onie-image.conf` as the data source. Any variable not defined in `onie-image.conf` can't be use here. That's why I added a new variable in `onie-image.conf`.